homeassistant/components/homewizard/coordinator.py

```diff
@@ -55,9 +61,19 @@ async def _async_update_data(self) -> DeviceResponseEntry:
                 data["system"] = await self.api.system()
 
         except RequestError as ex:
-            raise UpdateFailed("Device did not respond as expected") from ex
+            raise UpdateFailed(ex) from ex
 
         except DisabledError as ex:
-            raise UpdateFailed("API disabled, API must be enabled in the app") from ex
+            if not self.api_disabled:
+                self.api_disabled = True
+
+                # Do not reload when performing first refresh
+                if self.data is not None:
+                    await self.hass.config_entries.async_reload(self.entry_id)
+
+            raise UpdateFailed(ex) from ex
+
+        else:
```

Comment:
We don't need an `else:` block since we raise in the `except:` blocks above. We can remove this block and outdent the line below.